main/lsp/LSPTypechecker.h

```diff
@@ -200,13 +200,12 @@ class LSPTypecheckerInterface {
 };
 
 class LSPStaleTypechecker final : public LSPTypecheckerInterface {
+    std::shared_ptr<const LSPConfiguration> config;
     UndoState &undoState;
-
-    // Just so we have something to return from the stubbed functions after we ENFORCE(false)
-    ast::ParsedFile *pf;
+    std::unique_ptr<WorkerPool> emptyWorkers;
 
 public:
-    LSPStaleTypechecker(UndoState &undoState) : undoState(undoState), pf(nullptr) {}
+    LSPStaleTypechecker(std::shared_ptr<const LSPConfiguration> config, UndoState &undoState);
```

Comment:
It looks like there's an asymmetry between `LSPStaleTypechecker` and `LSPTypecheckerDelegate` here.

Why doesn't `LSPTypecheckerDelegate` take an `LSPConfiguration` or a `WorkerPool`?

Reply:
The `config` is used inside `LSPStaleTypechecker::query` directly because `LSPStaleTypechecker::query` largely duplicates the impelmentation of `LSPTypechecker::query`, while `LSPTypecheckerDelegate` can just delegate `LSPTypechecker::query`, where `config` is a field on `LSPTypechecker`.

The `WorkerPool` we create here exists for similar reasons: `LSPTypechecker` has a private `WorkerPool`. (Note that here we create an empty, i.e., zero-thread, `WorkerPool`, which actually causes work to run on the indexing thread itself.)